ansible_ai_connect/ai/api/model_pipelines/__init__.py

```diff
@@ -17,3 +20,26 @@
 import ansible_ai_connect.ai.api.model_pipelines.wca.pipelines_dummy  # noqa
 import ansible_ai_connect.ai.api.model_pipelines.wca.pipelines_onprem  # noqa
 import ansible_ai_connect.ai.api.model_pipelines.wca.pipelines_saas  # noqa
+from ansible_ai_connect.ai.api.model_pipelines.registry import REGISTRY
+from ansible_ai_connect.main.settings.types import t_model_mesh_api_type
+
+logger = logging.getLogger(__name__)
+
+
+def set_defaults(pipeline_provider):
+
+    def v_or_default(k, v):
+        defaults = REGISTRY["nop"]
+        if v is None:
+            logger.warning(
+                f"'{k.alias()}' is not available for provider '{pipeline_provider}',"
+                " failing back to 'nop'"
+            )
+            return defaults[k]
+        return v
+
+    return {k: v_or_default(k, v) for k, v in REGISTRY[pipeline_provider].items()}
+
+
```

Comment:
Since this runs during the module loading, it will be a bit tricky to properly unit-test it. I feel like it would be better to isolate this in a dedicated function.

Reply:
Looks like you're right (I've added a test but it's not being detected as covering this code).

I'll move it out.